src/SignalR/clients/cpp/src/signalrclient/http_client.h

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#pragma once
+
+#include <string>
+#include <functional>
+#include <map>
+#include <chrono>
+
+namespace signalr
+{
+    enum http_method
+    {
+        GET,
+        POST
+    };
+
+    class http_request
+    {
+    public:
+        http_method method;
+        std::map<std::string, std::string> headers;
+        std::string content;
+        std::chrono::seconds timeout;
```

Comment:
Maybe timeout should be a separate parameter on send. Or better yet, define a single-method CTS-like class that we return from send instead.